src/Mod/Sketcher/Gui/SketcherToolDefaultWidget.h

```diff
@@ -67,9 +67,11 @@ class SketcherToolDefaultWidget : public QWidget
     void setParameter(int parameterindex, double val);
     double getParameter(int parameterindex);
     bool isParameterSet(int parameterindex);
+    void updateVisualValue(int parameterindex, double val);
 
     void setParameterEnabled(int parameterindex, bool active = true);
     void setParameterFocus(int parameterindex);
+    void setParameterFont(int parameterindex, std::string fontStyle);

```

Comment:
If this function is intended so that the DrawSketch handler modifies the font of the widget, this is ok.

However, it appears that the intention is that it is an internal function of the widget. Then it should have a "private" access specifier.

Do not change anything at this time.

Reply:
As for the parameter, "fontStyle", it should be "const std::string &".

The reason is that here you are doing extra copies of the string when it is not necessary.

Passing a string by value only makes sense if you need to create a copy that you are going to consume (i.e. use it inside the function to call another function which takes a rvalue reference, i.e. std::move(fontStyle)).

An alternative is to make this function a consumer (by using an rvalue reference instead of a const reference). But this means that it must be called with std:move() and that the string is not valid after the call within the caller function.

Reply:
But now that I looked deeper into the body, I would advise against using a std::string at all. The reason is that creation of a string has a performance hit and comparison is not as efficient as a label (enum class). Here, I think a enum class is best for performance reasons. It is also better for the client, as a label is checked by the compiler (for errors in typing) as it must be a valid label. A string can silently fail just because a character is mistyped.

Do not change anything at this moment.

Reply:
I see. You're right an enum would do the exact same job. I should have used it instead of string.